unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/EditorMode/BuilderInWorldGodMode.cs

```diff
@@ -137,32 +155,89 @@ public override void Init(GameObject goToEdit, GameObject undoGo, GameObject sna
         HUDController.i.buildModeHud.OnScaleSelectedAction += ScaleMode;
     }
 
+    private void MouseDrag(int buttonId, Vector3 mousePosition, float axisX, float axisY)
+    {
+        if (buttonId != 0 ||
+            selectedEntities.Count <= 0)
+            return;
+
+        if(!isDraggingStarted)
+            StarDraggingSelectedEntities();
+
+        if (canDragSelectedEntities)
+        {
+            Vector3 currentPoint = GetFloorPointAtMouse(mousePosition);
+            Vector3 direction = currentPoint - dragStartedPoint;
+
+            editionGO.transform.position += direction;
+            dragStartedPoint = currentPoint;
+        }
+
+    }
+
     private void MouseUp(int buttonID, Vector3 position)
     {
-        if (mousePressed && buttonID == 0)
+        if (buttonID != 0)
+            return;
+
+        EndDraggingSelectedEntities();
+
+        if (isMakingSquareMultiSelection && mousePressed)
         {
-            if (isMakingSquareMultiSelection)
-            {
-                EndBoundMultiSelection();
-            }
+            EndBoundMultiSelection();
         }
     }
 
     void MouseDown(int buttonID, Vector3 position)
     {
-        if (buttonID == 0)
+        if (buttonID != 0)
+            return;
+
+        dragStartedPoint = GetFloorPointAtMouse(position);
+
+        if (squareMultiSelectionButtonPressed)
+        {
+            isMakingSquareMultiSelection = true;
+            isTypeOfBoundSelectionSelected = false;
+            isVoxelBoundMultiSelection = false;
```

Comment:
This is calling to have some kind of selection handler classes. Not for this PR but having this many toggles screams anti-pattern and too big class here.

Reply:
Yes, the voxel should have a separate mode, however, the product needs to give more specifications about how this is going to work. I will add it for the takeaways